### PR TITLE
Feature lua image orientation

### PR DESCRIPTION
**Pull request by rds13**
This PR is trying to provide a solution for #13686.

You can use this lua script to test it : https://gist.github.com/rds13/15675bd692133017d88cb18479abe40d

You need to 

1. Start darktable in lua debug mode with "-d lua" option
2. load the script from luarc, a Dummy exporter will be available
3. select some images in darktable
4. choose 'Dummy' as exporter
5. Export
6. You should see some trace with the path of images and the aspect_ratio found

### CODE REVIEW

**Comment by wpferguson** [comment]
Did you test this?

File: src/lua/image.c [hunk 1]
```diff
@@ -509,6 +509,9 @@ int dt_lua_init_image(lua_State *L)
   luaA_struct_member(L, dt_image_t, filename, const char_filename_length);
   luaA_struct_member(L, dt_image_t, width, const int32_t);
   luaA_struct_member(L, dt_image_t, height, const int32_t);
+  luaA_struct_member(L, dt_image_t, orientation, const int32_t);
+  luaA_struct_member(L, dt_image_t, aspect_ratio, float);
```
Comment (wpferguson) [inline src/lua/image.c:513]: You only want to read the value.  Exposing it this way allows it to be written.  For instance: You have a landscape photo and you get the aspect_ratio and it's 1.5.  You have a line in your script such as `image.aspect_ratio = 1.`.  The aspect_ratio is now set to square, but the image hasn't changed and still displays as landscape.  Just the sorting would be incorrect.

The better way to do this to add a get_aspect_ratio member function to read and return the value.

**Review by wpferguson — CHANGES_REQUESTED** [review]

File: src/lua/image.c (hunk 1 above)
Comment (wpferguson) [reply src/lua/image.c:513]: Tried declaring it as const and that works.  If you try and write to it, lua throws an error

```
     7.5432 LUA aspect ratio is 1.5
     7.5433 LUA ERROR : /home/bill/.config/dttest13695/lua/contrib/clear_GPS.lua:71: field "aspect_ratio" can't be written for type dt_lua_image_t

stack traceback:
	[C]: in ?
	[C]: in metamethod 'newindex'
	/home/bill/.config/dttest13695/lua/contrib/clear_GPS.lua:71: in upvalue 'clear_GPS'
	/home/bill/.config/dttest13695/lua/contrib/clear_GPS.lua:85: in function </home/bill/.config/dttest13695/lua/contrib/clear_GPS.lua:85>
	[C]: in ?


So, to sum up I'd be fine with losing the orientation and declaring aspect_ratio as const.
```

**Comment by rds13** [comment]
> Did you test this?

Yep, I was able to build it on macOS. It works from lua script. I should submit a lua script as an easy way to check the result.

File: src/lua/image.c [hunk 2]
```diff
@@ -509,6 +509,9 @@ int dt_lua_init_image(lua_State *L)
   luaA_struct_member(L, dt_image_t, filename, const char_filename_length);
   luaA_struct_member(L, dt_image_t, width, const int32_t);
   luaA_struct_member(L, dt_image_t, height, const int32_t);
+  luaA_struct_member(L, dt_image_t, orientation, const int32_t);
```
[the remark this answers is not in the document]
Comment (rds13) [reply src/lua/image.c:512]: I was thinking that the lua object should expose all properties of the dt_image_t. It was a step in that direction.
But I understand it should be polished with enum information.
At the moment I will get rid of it.

File: src/lua/image.c (hunk 1 above)
Comment (rds13) [reply src/lua/image.c:513]: I did'nt intend to have it as a read/write properties. I will fix that.

**Review by wpferguson — APPROVED** [review]

**Comment by wpferguson** [comment]
@TurboGit looks good to me.  Attached is a script to test it.  It adds a button to selected images.  It writes to the console.

[print_aspect_ratio.zip](https://github.com/darktable-org/darktable/files/10787796/print_aspect_ratio.zip)

**Comment by rds13** [comment]
Anything to add to API documentation ?

**Comment by wpferguson** [comment]
I'll take care of it once it's merged.  I need to do an update on the API docs anyway.

**Comment by TurboGit** [comment]
I found an issue, it prints the aspect ratio and then throw an error:

```
     4,7218 LUA aspect ratio is 1,5
     4,7221 LUA ERROR : ...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:67: field "aspect_ratio" can't be written for type dt_lua_image_t

stack traceback:
	[C]: in ?
	[C]: in metamethod 'newindex'
	...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:67: in upvalue 'print_aspect_ratio'
	...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:81: in function <...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:81>
	[C]: in ?
    13,2913 LUA aspect ratio is 0,89999997615814
    13,2915 LUA ERROR : ...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:67: field "aspect_ratio" can't be written for type dt_lua_image_t

stack traceback:
	[C]: in ?
	[C]: in metamethod 'newindex'
	...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:67: in upvalue 'print_aspect_ratio'
	...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:81: in function <...434/.config/darktable/lua/contrib/print_aspect_ratio.lua:81>
	[C]: in ?

```

**Review by TurboGit — APPROVED** [review]

**Review by TurboGit — CHANGES_REQUESTED** [review]
An exception is raised each time I click the button.

**Comment by TurboGit** [comment]
> The error is because the script is trying to set the value and it's read only. I was just testing to make sure it truly was read only.

So it works as designed :) Thanks.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by rds13** [comment]
👍 
Thank you @wpferguson for your patience and your kindness.

**Comment by wpferguson** [comment]
Thank you for fixing it
